src/Microsoft.AspNet.Http/HttpContextFactory.cs

```diff
@@ -17,13 +21,19 @@ public HttpContextFactory(IHttpContextAccessor httpContextAccessor)
         public HttpContext Create(IFeatureCollection featureCollection)
         {
             var httpContext = new DefaultHttpContext(featureCollection);
-            _httpContextAccessor.HttpContext = httpContext;
+            if (_httpContextAccessor != null)
+            {
+                _httpContextAccessor.HttpContext = httpContext;
```

Comment:
Does the null-conditional operator work on property setters? I.e. don't set the property if the instance is null. If not, this seems like a good example for a suggestion.